test/rlx_relup_SUITE.erl

```diff
@@ -176,3 +153,56 @@ make_relup_release2(Config) ->
     ?assert(lists:member({goal_app_1, "0.0.3"}, AppSpecs)),
     ?assert(lists:member({goal_app_2, "0.0.3"}, AppSpecs)),
     ?assert(lists:member({lib_dep_1, "0.0.1", load}, AppSpecs)).
+
+no_upfrom_release(Config) ->
+    LibDir1 = ?config(lib_dir, Config),
+
+    rlx_test_utils:create_app(LibDir1, "goal_app_1", "0.0.1", [stdlib,kernel,non_goal_1], []),
+    rlx_test_utils:create_app(LibDir1, "goal_app_1", "0.0.2", [stdlib,kernel,non_goal_1], []),
+    {ok, GA1} = rlx_test_utils:create_app(LibDir1, "goal_app_1", "0.0.3", [stdlib,kernel,non_goal_1], []),
+    rlx_test_utils:create_app(LibDir1, "lib_dep_1", "0.0.1", [stdlib,kernel], []),
+    rlx_test_utils:create_app(LibDir1, "goal_app_2", "0.0.1", [stdlib,kernel,goal_app_1,non_goal_2], []),
+    rlx_test_utils:create_app(LibDir1, "goal_app_2", "0.0.2", [stdlib,kernel,goal_app_1,non_goal_2], []),
+    {ok, GA2} = rlx_test_utils:create_app(LibDir1, "goal_app_2", "0.0.3", [stdlib,kernel,goal_app_1,non_goal_2], []),
+    rlx_test_utils:create_app(LibDir1, "non_goal_1", "0.0.1", [stdlib,kernel], [lib_dep_1]),
+    rlx_test_utils:create_app(LibDir1, "non_goal_2", "0.0.1", [stdlib,kernel], []),
+
+    rlx_test_utils:write_appup_file(GA1, "0.0.2"),
+    rlx_test_utils:write_appup_file(GA2, "0.0.2"),
+
+    RelxConfig = [{release, {foo, "0.0.1"},
+                   [sasl,
+                    {goal_app_1, "0.0.1"},
+                    {goal_app_2, "0.0.1"}]},
+                  {release, {foo, "0.0.2"},
+                   [sasl,
+                    {goal_app_1, "0.0.2"},
+                    {goal_app_2, "0.0.2"}]},
+                  {release, {foo, "0.0.3"},
+                   [sasl,
+                    {goal_app_1, "0.0.3"},
+                    {goal_app_2, "0.0.3"}]}],
+    OutputDir = filename:join([?config(priv_dir, Config),
+                               rlx_test_utils:create_random_name("relx-output")]),
+
+    Apps = rlx_test_utils:all_apps([LibDir1]),
+    %% {ok, _State} = relx:build_release({foo, "0.0.1"}, Apps, [{root_dir, LibDir1},
+    %%                                                          {output_dir, OutputDir} | RelxConfig]),
+
+    %% {ok, _State1} = relx:build_release({foo, "0.0.2"}, Apps, [{root_dir, LibDir1},
+    %%                                                           {output_dir, OutputDir} | RelxConfig]),
+
+    %% %% Goal apps are removed to simulate a users dev environment where the apps
+    %% %% being used in an appup/relup are likely only under _rel/<release>/lib/
+    %% ec_file:remove(filename:join(LibDir1, "goal_app_1-0.0.1"), [recursive]),
+    %% ec_file:remove(filename:join(LibDir1, "goal_app_1-0.0.2"), [recursive]),
```

Comment:
Same question re: commented tests

Reply:
This was just forgetting to go back and clean up. Removing that code now. It is a new test I added, it didn't exist in relx before.